### ISSUE TITLE
Tests fail during build on Centos 6

### ISSUE THREAD

**Issue by minusdavid** [net-amqp-rabbitmq/net-amqp-rabbitmq#204]
I know that Centos 6 is ancient and I'm in the process of replacing it with an up-to-date version, but I'm just flagging it anyway.

```
+ /usr/bin/perl Makefile.PL INSTALLDIRS=vendor 'OPTIMIZE=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic'
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Net::AMQP::RabbitMQ
Writing MYMETA.yml and MYMETA.json
+ make -j2
Running Mkbootstrap for RabbitMQ ()
chmod 644 "RabbitMQ.bs"
"/usr/bin/perl" "/usr/share/perl5/ExtUtils/xsubpp"  -typemap '/usr/share/perl5/ExtUtils/typemap' -typemap '/root/rpmbuild/BUILD/Net-AMQP-RabbitMQ-2.40008/typemap'  RabbitMQ.xs > RabbitMQ.xsc
cp RabbitMQ.pm blib/lib/Net/AMQP/RabbitMQ.pm
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_api.c
mv RabbitMQ.xsc RabbitMQ.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_connection.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_consumer.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_framing.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_hostcheck.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_mem.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_openssl.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_openssl_bio.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_openssl_hostname_validation.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_socket.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_table.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_tcp_socket.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_time.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L amqp_url.c
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L perl_math_int64.c
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- RabbitMQ.bs blib/arch/auto/Net/AMQP/RabbitMQ/RabbitMQ.bs 644
gcc -c  -I. -Irabbitmq-include -Irabbitmq-include/unix -D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fe
xceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic   -DVERSION=\"2.40008\" -DXS_VERSION=\"2.40008\" -fPIC "-I/usr/lib64/perl5/CORE"  -DHAVE_SOCKET -DHAVE_POLL -DAMQ_PLATFORM='"Perl_linux_v5.10.1"' -DNAR_HAVE_OPENSS
L RabbitMQ.c
In file included from perl_math_int64.c:12:
ppport.h:6145:1: warning: "MUTABLE_PTR" redefined
In file included from /usr/lib64/perl5/CORE/perl.h:2424,
                 from perl_math_int64.c:11:
/usr/lib64/perl5/CORE/handy.h:63:1: warning: this is the location of the previous definition
In file included from RabbitMQ.xs:10:
ppport.h:6145:1: warning: "MUTABLE_PTR" redefined
In file included from /usr/lib64/perl5/CORE/perl.h:2424,
                 from RabbitMQ.xs:2:
/usr/lib64/perl5/CORE/handy.h:63:1: warning: this is the location of the previous definition
RabbitMQ.xs: In function 'read_message':
RabbitMQ.xs:358: warning: value computed is not used
RabbitMQ.xs:361: warning: value computed is not used
RabbitMQ.xs:377: warning: value computed is not used
RabbitMQ.xs:380: warning: value computed is not used
RabbitMQ.xs:383: warning: value computed is not used
RabbitMQ.xs:386: warning: value computed is not used
RabbitMQ.xs:389: warning: value computed is not used
RabbitMQ.xs:392: warning: value computed is not used
RabbitMQ.xs:395: warning: value computed is not used
RabbitMQ.xs:398: warning: value computed is not used
RabbitMQ.xs:401: warning: value computed is not used
RabbitMQ.xs:404: warning: value computed is not used
RabbitMQ.xs:409: warning: value computed is not used
RabbitMQ.xs:429: warning: value computed is not used
RabbitMQ.xs:438: warning: value computed is not used
RabbitMQ.xs:446: warning: value computed is not used
RabbitMQ.xs:454: warning: value computed is not used
RabbitMQ.xs:462: warning: value computed is not used
RabbitMQ.xs:470: warning: value computed is not used
RabbitMQ.xs:478: warning: value computed is not used
RabbitMQ.xs:486: warning: value computed is not used
RabbitMQ.xs:494: warning: value computed is not used
RabbitMQ.xs:503: warning: value computed is not used
RabbitMQ.xs:513: warning: value computed is not used
RabbitMQ.xs:531: warning: value computed is not used
RabbitMQ.xs:552: warning: value computed is not used
RabbitMQ.xs:561: warning: value computed is not used
RabbitMQ.xs: In function 'consume_message':
RabbitMQ.xs:689: warning: value computed is not used
RabbitMQ.xs:690: warning: value computed is not used
RabbitMQ.xs:691: warning: value computed is not used
RabbitMQ.xs:692: warning: value computed is not used
RabbitMQ.xs:693: warning: value computed is not used
RabbitMQ.xs:694: warning: value computed is not used
RabbitMQ.xs:701: warning: value computed is not used
RabbitMQ.xs:702: warning: value computed is not used
RabbitMQ.xs: In function 'mq_table_to_hashref':
RabbitMQ.xs:970: warning: value computed is not used
RabbitMQ.xs: In function 'basic_get':
RabbitMQ.xs:1100: warning: value computed is not used
RabbitMQ.xs:1101: warning: value computed is not used
RabbitMQ.xs:1102: warning: value computed is not used
RabbitMQ.xs:1103: warning: value computed is not used
RabbitMQ.xs:1104: warning: value computed is not used
RabbitMQ.xs:1111: warning: value computed is not used
RabbitMQ.xs:1112: warning: value computed is not used
RabbitMQ.c: In function 'XS_Net__AMQP__RabbitMQ_recv':
RabbitMQ.xs:1588: warning: unused variable 'envelope'
RabbitMQ.c: In function 'XS_Net__AMQP__RabbitMQ__publish':
RabbitMQ.xs:1689: warning: unused variable 'exchange'
RabbitMQ.c: In function 'XS_Net__AMQP__RabbitMQ_disconnect':
RabbitMQ.xs:1861: warning: unused variable 'sockfd'
RabbitMQ.c: In function 'XS_Net__AMQP__RabbitMQ__new':
RabbitMQ.c:2705: warning: unused variable 'clazz'
RabbitMQ.xs: In function 'read_message':
RabbitMQ.xs:318: warning: 'body_sv' may be used uninitialized in this function
rm -f blib/arch/auto/Net/AMQP/RabbitMQ/RabbitMQ.so
LD_RUN_PATH="/lib64:/usr/lib64" gcc  -shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic  RabbitMQ.o amqp_api.o amqp_connection.o amqp_consumer.o amqp_framing.o
amqp_hostcheck.o amqp_mem.o amqp_openssl.o amqp_openssl_bio.o amqp_openssl_hostname_validation.o amqp_socket.o amqp_table.o amqp_tcp_socket.o amqp_time.o amqp_url.o perl_math_int64.o   -o blib/arch/auto/Net/AMQP/RabbitMQ/RabbitMQ.so  \
           -lrt -lssl -lcrypto   \

chmod 755 blib/arch/auto/Net/AMQP/RabbitMQ/RabbitMQ.so
Manifying 1 pod document
+ exit 0
Executing(%install): /bin/sh -e /var/tmp/rpm-tmp.ZQxNKJ
+ umask 022
+ cd /root/rpmbuild/BUILD
+ '[' /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64 '!=' / ']'
+ rm -rf /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64
++ dirname /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64
+ mkdir -p /root/rpmbuild/BUILDROOT
+ mkdir /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64
+ cd Net-AMQP-RabbitMQ-2.40008
+ LANG=C
+ export LANG
+ unset DISPLAY
+ rm -rf /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64
+ make pure_install PERL_INSTALL_ROOT=/root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- RabbitMQ.bs blib/arch/auto/Net/AMQP/RabbitMQ/RabbitMQ.bs 644
Manifying 1 pod document
Files found in blib/arch: installing files in blib/lib into architecture dependent library tree
Installing /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64/usr/lib64/perl5/vendor_perl/auto/Net/AMQP/RabbitMQ/RabbitMQ.so
Installing /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64/usr/lib64/perl5/vendor_perl/Net/AMQP/RabbitMQ.pm
Installing /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64/usr/share/man/man3/Net::AMQP::RabbitMQ.3pm
+ find /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64 -type f -name .packlist -exec rm -f '{}' ';'
+ find /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64 -type f -name '*.bs' -size 0 -exec rm -f '{}' ';'
+ find /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64 -depth -type d -exec rmdir '{}' ';'
+ /bin/chmod -Rf a+rX,u+w,g-w,o-w /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64/usr
+ /usr/lib/rpm/find-debuginfo.sh --strict-build-id /root/rpmbuild/BUILD/Net-AMQP-RabbitMQ-2.40008
extracting debug info from /root/rpmbuild/BUILDROOT/perl-Net-AMQP-RabbitMQ-2.40008-1.el6.x86_64/usr/lib64/perl5/vendor_perl/auto/Net/AMQP/RabbitMQ/RabbitMQ.so
1186 blocks
+ /usr/lib/rpm/check-buildroot
+ /usr/lib/rpm/redhat/brp-compress
+ /usr/lib/rpm/redhat/brp-strip-static-archive /usr/bin/strip
+ /usr/lib/rpm/redhat/brp-strip-comment-note /usr/bin/strip /usr/bin/objdump
+ /usr/lib/rpm/brp-python-bytecompile /usr/bin/python
+ /usr/lib/rpm/redhat/brp-python-hardlink
+ /usr/lib/rpm/redhat/brp-java-repack-jars
Executing(%check): /bin/sh -e /var/tmp/rpm-tmp.LfKT8G
+ umask 022
+ cd /root/rpmbuild/BUILD
+ cd Net-AMQP-RabbitMQ-2.40008
+ unset DISPLAY
+ make test
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- RabbitMQ.bs blib/arch/auto/Net/AMQP/RabbitMQ/RabbitMQ.bs 644
PERL_DL_NONLAZY=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/000_use_ok.t ......................... ok
t/001_declare_exchange.t ............... ok
t/002_publish_consume.t ................ 13/21
#   Failed test 'large payload matches'
#   at t/002_publish_consume.t line 101.
#     Structures begin differing at:
#          $got = undef
#     $expected = HASH(0x10e67a8)
t/002_publish_consume.t ................ 21/21 # Looks like you failed 1 test of 21.
t/002_publish_consume.t ................ Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/21 subtests
t/005_noack.t .......................... ok
t/006_txn.t ............................ ok
t/007_get.t ............................ ok
t/008_queue_declare.t .................. ok
t/009_no_route.t ....................... ok
t/010_qos.t ............................ ok
t/011_hearbeat.t ....................... 18/18
#   Failed test 'publish fails'
#   at t/011_hearbeat.t line 40.
# Looks like you failed 1 test of 18.
t/011_hearbeat.t ....................... Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/18 subtests
t/012_timeout.t ........................ ok
t/013_headers.t ........................ ok
t/014_bind_with_headers.t .............. ok
t/015_declare_queue_with_headers.t ..... ok
t/016_reject.t ......................... ok
t/018_is_connected.t ................... ok
t/019_utf8_safety.t .................... ok
t/020_dead_channel_close.t ............. ok
t/021_json_utf8_in_and_out.t ........... ok
t/022_array_headers.t .................. ok
t/023_numeric_header_fields.t .......... ok
t/024_boolean_header_fields.t .......... ok
t/025_declare_exchange_with_ae.t ....... ok
t/026_getsockfd_and_get_channel_max.t .. ok
t/027_bad_socket_detection.t ........... ok
t/028_exchange_to_exchange.t ........... ok
t/029_amqp_properties.t ................ ok
t/030_ssl.t ............................ ok
t/031_magic_header_fields.t ............ ok
t/032_cancel_consume_ack_exception.t ... ok
t/033_fork.t ........................... ok

Test Summary Report
-------------------
t/002_publish_consume.t              (Wstat: 256 Tests: 21 Failed: 1)
  Failed test:  13
  Non-zero exit status: 1
t/011_hearbeat.t                     (Wstat: 256 Tests: 18 Failed: 1)
  Failed test:  18
  Non-zero exit status: 1
Files=31, Tests=348, 131 wallclock secs ( 0.20 usr  0.05 sys +  1.09 cusr  0.29 csys =  1.63 CPU)
Result: FAIL
```

**Comment by manchicken** [comment]
I'd need to know more details on this to troubleshoot it more. Were you running RabbitMQ on the same host? Can you give me the `perl -V`?

If you could get a container to fail in the same way that would be super helpful. Outside of that, I have no idea how to troubleshoot this.

**Comment by minusdavid** [comment]
I actually signed on just now to tell the OP that Centos 6 is EOL and that you shouldn't bother trying to resolve this one, and then I noticed I was the OP haha. 

I suppose technically RHEL 6 has its ELS support until June 30 2024, but...

Anyway, the Perl version is that special Red Hat Perl 5.10.1 I believe. 

I did my work in a container using the centos6 image (https://hub.docker.com/_/centos) although it's so outdated that the yum repos aren't available anymore, so /etc/yum.repos.d/CentOS-Base.repo has to be replaced with one that points to http://vault.centos.org/6.10/* URLs.

Probably more work than its worth.
